lib/ctblmono.gi

```diff
@@ -2176,6 +2176,74 @@ InstallMethod( IsMinimalNonmonomial,
     return false;
 end );
 
+InstallMethod( IsMinimalNonmonomial,
+    "for a non-solvable group",
+    [ IsGroup ],
+    function( K )
+
+    local info, q, p, f;
+
+    if IsSolvableGroup(K) then
+      TryNextMethod();
+    fi;
+
+    # A non-solvable group all of whose proper quotients are monomial
+    # must have every proper quotient solvable. Hence G must be simple.
+    if not IsSimpleGroup(K) then
+      return false;
+    fi;
+
+    # If in addition, every proper subgroup is monomial, then every
+    # proper subgroup must be solvable, so G must be minimal simple.
+    #
+    # A minimal non-simple group must be (by Thompson's N-Groups paper):
+    # * PSL(2,2^f) for f prime, and all such are minimal-non-M groups
+    # * PSL(2,3^f) for f an odd prime, and all such are minimal-non-M groups
+    # * PSL(3,3) which is NOT a minimal-non-M group (contains SL(2,3))
+    # * Sz(2^f) for f an odd prime, and all such are minimal-non-M groups
```

Comment:
That list seems to be missing the groups PSL(2,p) for p > 3 a prime congruent to 2 or 3 mod 5?

Reply:
I've updated the comments now too. The code was already updated as per your suggestion